packages/solidity-contracts/contracts/messaging/gateway/FuelERC20Gateway/FuelERC20GatewayV4.sol

```diff
@@ -322,6 +325,68 @@ contract FuelERC20GatewayV4 is
         emit Withdrawal(bytes32(uint256(uint160(to))), tokenAddress, amount);
     }
 
+    /// @notice Deposits the given tokens to an account on Fuel using permit signature for approval
+    /// @param to Fuel address to deposit tokens to
+    /// @param tokenAddress Address of the token being transferred to Fuel
+    /// @param amount Amount of tokens to deposit
+    /// @param permitSignature struct containing the permit signature
+    /// @dev Made payable to reduce gas costs
+    function depositWithPermit(
+        bytes32 to,
+        address tokenAddress,
+        uint256 amount,
+        PermitSignature memory permitSignature
+    ) public payable virtual whenNotPaused {
+        // If the deadline is zero, we revert as we assume the token does not have permit functionality so then `deposit` can be called directly.
+        if (permitSignature.deadline == 0) revert PermitNotAllowed();
+
+        // sets token allowance with permit signature
+        IERC20PermitUpgradeable(tokenAddress).permit(
```

Comment:
hm, yeah actually since there is a fallback to call the existing non-permit method so that's why probably didn't do it, but yeah I think we can add this in